cmake/python.cmake

```diff
@@ -10,32 +10,26 @@ message(STATUS "Using PYTHON_EXECUTABLE: ${PYTHON_EXECUTABLE}")
 
 set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
 
-set(enable_setuptools_deb_layout OFF)
-if(EXISTS "/etc/debian_version")
-  set(enable_setuptools_deb_layout ON)
+# setuptools is fussy about windows paths, make sure the install prefix is in native format
+if(WIN32)
+  file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
 endif()
-option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
 
-if(SETUPTOOLS_DEB_LAYOUT)
-  message(STATUS "Using Debian Python package layout")
-  set(PYTHON_PACKAGES_DIR dist-packages)
-  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
-  # use major version only when installing 3.x with debian layout
-  if("${PYTHON_VERSION_MAJOR}" STREQUAL "3")
-    set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}")
+# this block determines the same value as the Python function get_python_install_dir() from python/catkin/builder.py
+if(NOT DEFINED PYTHON_INSTALL_DIR)
+  execute_process(COMMAND "${PYTHON_EXECUTABLE}"
+    "-c" "from distutils.sysconfig import get_python_lib; import os; print(os.sep.join(get_python_lib().split(os.sep)[-2 if os.name == 'nt' else -3:]))"
+    RESULT_VARIABLE _res
+    OUTPUT_VARIABLE _var
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(NOT _res EQUAL 0)
+    message(FATAL_ERROR "Determine PYTHON_INSTALL_DIR failed")
   endif()
-else()
-  message(STATUS "Using default Python package layout")
-  set(PYTHON_PACKAGES_DIR site-packages)
-  # setuptools is fussy about windows paths, make sure the install prefix is in native format
-  file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
+  set(PYTHON_INSTALL_DIR "${_var}"
+    CACHE INTERNAL "This needs to be in PYTHONPATH when 'setup.py install' is called.  And it needs to match.  But setuptools won't tell us where it will install things.")
```

Comment:
It needs to match what? And what purpose does the stuff about setuptools serve. I don't understand these parts of the variable description. I think it either needs to be more specific or maybe just leave this stuff out.